### PR TITLE
[lz4] Update to 1.10.0, use official CMake build system

### PR DESCRIPTION
**Pull request by dg0yt**
Based on #40021, with ~a tentative~ an accepted change for https://github.com/lz4/lz4/issues/1179:
Keeping the `lz4::lz4` target for CMake config package `lz4`, i.e. consistent capitalization.

### CODE REVIEW

File: ports/lz4/portfile.cmake [hunk 1]
```diff
@@ -1,22 +1,36 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO lz4/lz4
-    REF v1.9.4
-    SHA512 043a9acb2417624019d73db140d83b80f1d7c43a6fd5be839193d68df8fd0b3f610d7ed4d628c2a9184f7cde9a0fd1ba9d075d8251298e3eb4b3a77f52736684
+    REF v${VERSION}
+    SHA512 8c4ceb217e6dc8e7e0beba99adc736aca8963867bcf9f970d621978ba11ce92855912f8b66138037a1d2ae171e8e17beb7be99281fea840106aa60373c455b28
     HEAD_REF dev
+    PATCHES
+        target-lz4-lz4.diff
```
Comment (dg0yt) [inline ports/lz4/portfile.cmake:8]: https://github.com/lz4/lz4/pull/1470

File: ports/lz4/target-lz4-lz4.diff [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+index e9c9473..4b48032 100644
+--- a/build/cmake/lz4Config.cmake.in
++++ b/build/cmake/lz4Config.cmake.in
+@@ -1,2 +1,10 @@
+ @PACKAGE_INIT@
+-include( "${CMAKE_CURRENT_LIST_DIR}/lz4Targets.cmake" )
+\ No newline at end of file
++include( "${CMAKE_CURRENT_LIST_DIR}/lz4Targets.cmake" )
++if(NOT TARGET lz4::lz4)
++    add_library(lz4::lz4 INTERFACE IMPORTED)
```
Comment (teo-tsirpanis) [inline ports/lz4/target-lz4-lz4.diff:11]: You can just link to `LZ4::lz4`.

Comment (dg0yt) [reply ports/lz4/target-lz4-lz4.diff:11]: I don't know which point you are trying to make for the `add_library` line.
(This PR, with https://github.com/lz4/lz4/pull/1470, is about never using `LZ4::lz4`.)

File: ports/lz4/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/lz4/portfile.cmake:8]: Merged upstream :tada:

**Comment by MonicaLiu0311** [comment]
lz4
-
All features are tested successfully in the following triplet:
```
x86-windows
x64-windows
x64-windows-static
```

**Review by MonicaLiu0311 — APPROVED** [review]

**Review by data-queue — APPROVED** [review]

**Comment by data-queue** [comment]
Thanks @teo-tsirpanis and @dg0yt for the update!

**Pull request merged**